src/lib/cache/subCache.cpp

```diff
@@ -99,6 +100,26 @@ entityId(_entityId), entityType(_entityType)
   {
     entityIdPatternToBeFreed = false;
   }
+
+  if (isTypePattern)
+  {
+    // FIXME P5: recomp error should be captured? have a look to other usages of regcomp()
+    // in order to see how it works
+    if (regcomp(&entityTypePattern, _entityType.c_str(), REG_EXTENDED) != 0)
+    {
+      alarmMgr.badInput(clientIp, "invalid regular expression for typePattern");
+      isTypePattern = false;  // FIXME P6: this entity should not be let into the system. Must be stopped before.
+                          //           Right here, best thing to do is simply to say it is not a regex
```

Comment:
As entityTypePatternToBeFreed seems to be set everywhere, it should probably be set here too (to FALSE).
However, it would be better to set it to FALSE in the constructor and only set to to TRUE where needed.